app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -1423,6 +1423,11 @@ public void onPlayerError(@NonNull final PlaybackException error) {
             case ERROR_CODE_TIMEOUT:
             case ERROR_CODE_IO_UNSPECIFIED:
             case ERROR_CODE_IO_NETWORK_CONNECTION_FAILED:
```

Comment:
As you can see on [the JavaDoc of the error code](https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/PlaybackException.html#ERROR_CODE_IO_NETWORK_CONNECTION_FAILED), using this error and show to the user that it doesn't have a connection in this case is not appropriate for all cases.

I recommend checking the network connection using the system APIs / compat/JetPack APIs if there are ones instead.